django/core/serializers/python.py

```diff
@@ -142,12 +145,17 @@ def m2m_convert(v):
                         return force_text(model._meta.pk.to_python(v), strings_only=True)
 
                 try:
-                    m2m_data[field.name] = []
+                    m2m_field_values = []
                     for pk in field_value:
-                        m2m_data[field.name].append(m2m_convert(pk))
+                        m2m_field_values.append(m2m_convert(pk))
                 except Exception as e:
+                    if isinstance(e, ObjectDoesNotExist) and handle_forward_references:
```

Comment:
Agree it's not very nice -- but we only want to handle `ObjectDoesNotExist` when `handle_forward_references` is `True`.  This goes away if we decide we do always want to handle forward references.